src/pages/ingredient/IngredientPage.tsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useParams } from "react-router-dom";
+import { AppDispatch } from "../../services/Store";
+import { selectFetchedIngredients } from "../../services/reducers/IngredientsReducer";
+import { IngredientType } from "../../types/Ingredient.type";
+import { Nullable } from "../../types/common.type";
+import IngredientDetails from "../../components/modal/ingredients/IngredientDetails";
+import { useTranslation } from "react-i18next";
+import styles from "./IngredientPage.module.css";
+
+const IngredientPage = () => {
+  let { ingredientId } = useParams();
+  const { t } = useTranslation("ingredients");
+  const ingredients = useSelector(selectFetchedIngredients);
+  const [ingredient, setIngredient] = useState<Nullable<IngredientType>>(null);
```

Comment:
Надо исправить: необхоидмо типизировать `useSelector`. К примеру вот пример 
```js
import type { TypedUseSelectorHook } from 'react-redux'
import { useDispatch, useSelector, useStore } from 'react-redux'
import type { AppDispatch, AppStore, RootState } from './store'

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export const useAppStore: () => AppStore = useStore
```

Подробнее https://react-redux.js.org/using-react-redux/usage-with-typescript

Reply:
использовал новый сетап, спасибо!